src/NServiceBus.AzureStoragePersistence.AcceptanceTests/When_saga_is_started_by_two_types_of_messages.cs

```diff
@@ -0,0 +1,143 @@
+                    .Run();
+
+            CollectionAssert.AreEquivalent(guids, context.CompletedIds.ToArray());
+        }
+
+        public class ProvideConfiguration : IProvideConfiguration<TransportConfig>
+        {
+            public TransportConfig GetConfiguration()
+            {
+                return new TransportConfig
+                {
+                    MaximumConcurrencyLevel = 3,
+                };
+            }
+        }
+
+        public class Context : ScenarioContext
+        {
+            private readonly ConcurrentDictionary<string, string> completed = new ConcurrentDictionary<string, string>();
+            public int CompletedIdsCount => completed.Count;
+            public IEnumerable<string> CompletedIds => completed.Keys;
+
+            public void MarkAsCompleted(string orderId)
+            {
+                completed.AddOrUpdate(orderId, orderId, (o1, o2) => o1);
+            }
+        }
+
+        public class ReceiverWithSagas : EndpointConfigurationBuilder
+        {
+            public ReceiverWithSagas()
+            {
+                EndpointSetup<DefaultServer>(
+                    config =>
+                    { });
+            }
+        }
+
+        public class ShippingPolicy : Saga<ShippingPolicy.State>,
+            IAmStartedByMessages<OrderPlaced>,
+            IAmStartedByMessages<OrderBilled>
+        {
+            public Context Context { get; set; }
+
+            public void Handle(OrderBilled message)
+            {
+                Data.OrderId = message.OrderId;
+                Data.Billed = true;
+
+                TryComplete();
+            }
+
+            public void Handle(OrderPlaced message)
+            {
+                Data.OrderId = message.OrderId;
+                Data.Placed = true;
+
+                TryComplete();
+            }
+
+            private void TryComplete()
+            {
+                if (Data.Billed && Data.Placed)
+                {
+                    MarkAsComplete();
+                    Context.MarkAsCompleted(Data.OrderId);
+                }
+            }
+
+            protected override void ConfigureHowToFindSaga(SagaPropertyMapper<State> mapper)
+            {
+                mapper.ConfigureMapping<OrderPlaced>(m => m.OrderId)
+                    .ToSaga(s => s.OrderId);
+                mapper.ConfigureMapping<OrderBilled>(m => m.OrderId)
+                    .ToSaga(s => s.OrderId);
+            }
+
+            public class State : IContainSagaData
+            {
+                [Unique]
+                public virtual string OrderId { get; set; }
+                public virtual bool Placed { get; set; }
+                public virtual bool Billed { get; set; }
+                public virtual Guid Id { get; set; }
+                public virtual string Originator { get; set; }
+                public virtual string OriginalMessageId { get; set; }
+            }
+        }
+
+
+        [Serializable]
```

Comment:
It's core v5. I can't remove it but will clean up as we align ASP to the core v6.